scripts/create_newcase

```diff
@@ -4,6 +4,9 @@ from Tools.standard_script_setup import *
 from CIME.utils         import expect, get_model
 from CIME.case          import Case
 from CIME.check_lockedfiles import lock_file
+import re
+
+invalid_case_name_re = re.compile(r"[\.<>/{}[\]~`@]");
```

Comment:
periods should definitely be allowed in the case name - e.g., they are used extensively in the case name for tests. But they should not be allowed in the testid, if I remember correctly